### PR TITLE
Update checkout action version

### PR DESCRIPTION
**Pull request by ncclementi**
The macos ci runs for py3.8 and 3.10 keep hanging. Trying out couple of things here and see what happens.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
 
     steps:
       - name: Checkout source
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3.5.2
```
Comment (j-bennet) [inline .github/workflows/tests.yml:28]: Why not `v3`, like it advises here?
https://github.com/marketplace/actions/checkout
Do we have to be explicit? Not a blocking comment, just curious.

Comment (ncclementi) [reply .github/workflows/tests.yml:28]: I was following what we have in dask/dask but we can just do v3 and see what happens

Comment (ncclementi) [inline .github/workflows/tests.yml:28]:
```suggestion
        uses: actions/checkout@v3
```

**Comment by j-bennet** [comment]
I see that all macos jobs have finished, did you have to restart some?

**Comment by ncclementi** [comment]
>I see that all macos jobs have finished, did you have to restart some?

No I didn't :tada:. I'll merge this and see what happens.

**Pull request merged**